forge-ai/src/main/java/forge/ai/ability/TokenAi.java

```diff
@@ -69,15 +69,19 @@ protected boolean checkPhaseRestrictions(final Player ai, final SpellAbility sa,
 
         Card actualToken = spawnToken(ai, sa);
 
-        if (actualToken == null || (actualToken.isCreature() && actualToken.getNetToughness() < 1)) {
-            // planeswalker plus ability or sub-ability is useful
-            return pwPlus || sa.getSubAbility() != null;
-        }
-
         String tokenAmount = sa.getParamOrDefault("TokenAmount", "1");
         String tokenPower = sa.getParamOrDefault("TokenPower", actualToken.getBasePowerString());
         String tokenToughness = sa.getParamOrDefault("TokenToughness", actualToken.getBaseToughnessString());
 
+        // Don't check toughness yet if token has variable P/T based on X
+        // (the check will happen after X is calculated)
+        boolean hasVariablePT = "X".equals(tokenAmount) || "X".equals(tokenPower) || "X".equals(tokenToughness);
+
+        if (!hasVariablePT && (actualToken == null || (actualToken.isCreature() && actualToken.getNetToughness() < 1))) {
```

Comment:
thanks for contributing

I think it's probably ok for AI to behave somewhat less greedy like this

you can also reuse your new boolean in L86
and maybe change its name to make it slightly more precise